File: modulesCore/loginAndRegister/src/main/java/cn/pivotstudio/modulec/loginandregister/repository/LoginRepository.java
package cn.pivotstudio.modulec.loginandregister.repository;

import android.annotation.SuppressLint;
import androidx.lifecycle.MutableLiveData;
import cn.pivotstudio.husthole.moduleb.network.BaseObserver;
import cn.pivotstudio.husthole.moduleb.network.NetworkApi;
import cn.pivotstudio.husthole.moduleb.network.errorhandler.ExceptionHandler;
import cn.pivotstudio.modulec.loginandregister.model.LoginResponse;
import cn.pivotstudio.modulec.loginandregister.network.LRRequestInterface;
import java.util.HashMap;

/**
 * @classname:LoginRepository
 * @description:
 * @date:2022/4/29 14:20
 * @version:1.0
 * @author:
 */
@SuppressLint("CheckResult")
public class LoginRepository {
    public final MutableLiveData<String> failed = new MutableLiveData<>();
    public MutableLiveData<LoginResponse> login = new MutableLiveData<>();

    //  private final MMKVUtil mvUtil;
    // @Inject
    public LoginRepository() {

        //获取mvUtils
        //   MMKVUtilEntryPoint entryPoint =
        //        EntryPointAccessors.fromApplication(getContext(), MMKVUtilEntryPoint.class);
        //   mvUtil = entryPoint.getMVUtils();
    }

    public void getLoginForNetwork(String id, String password) {
        HashMap<String, String> map = new HashMap<>();
        map.put("email", id);
        map.put("password", password);
        NetworkApi.createService(LRRequestInterface.class, 2)
            .mobileLogin(map)
            .compose(NetworkApi.applySchedulers(new BaseObserver<LoginResponse>() {
                @Override
                public void onSuccess(LoginResponse loginResponse) {
                    login.postValue(loginResponse);
                }

                @Override
                public void onFailure(Throwable e) {
                    failed.postValue(((ExceptionHandler.ResponseThrowable) e).message);
                }
            }));
    }
}

File: modulesCore/loginAndRegister/src/main/java/cn/pivotstudio/modulec/loginandregister/repository/LoginRepository.kt
package cn.pivotstudio.modulec.loginandregister.repository

import android.annotation.SuppressLint
import androidx.lifecycle.MutableLiveData
import cn.pivotstudio.modulec.loginandregister.model.LoginResponse
import cn.pivotstudio.husthole.moduleb.network.NetworkApi
import cn.pivotstudio.modulec.loginandregister.network.LRRequestInterface
import cn.pivotstudio.husthole.moduleb.network.BaseObserver
import cn.pivotstudio.husthole.moduleb.network.errorhandler.ExceptionHandler.ResponseThrowable
import cn.pivotstudio.modulec.loginandregister.network.LoginAndRegisterNetworkApi
import java.util.HashMap

/**
 * @classname:LoginRepository
 * @description:
 * @date:2022/4/29 14:20
 * @version:1.0
 * @author:
 */
@SuppressLint("CheckResult")
class LoginRepository  //  private final MMKVUtil mvUtil;
// @Inject
{
    @JvmField
    val failed = MutableLiveData<String>()

    @JvmField
    var login = MutableLiveData<LoginResponse>()
    fun getLoginForNetwork(id: String, password: String) {
        val map = HashMap<String, String>()
        map["email"] = id
        map["password"] = password
        LoginAndRegisterNetworkApi.retrofitService
            .mobileLogin(map)
            .compose(NetworkApi.applySchedulers(object : BaseObserver<LoginResponse>() {
                override fun onSuccess(loginResponse: LoginResponse) {
                    login.postValue(loginResponse)
                }

                override fun onFailure(e: Throwable) {
                    failed.postValue((e as ResponseThrowable).message)
                }
            }))
    }
}